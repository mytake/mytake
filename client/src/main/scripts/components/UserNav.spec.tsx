/*
 * MyTake.org website and tooling.
 * Copyright (C) 2018-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import React from "react";
import renderer from "react-test-renderer";
import UserNav from "./UserNav";

jest.mock("./DropDown", () => ({
  __esModule: true,
  default: "DropDown",
}));

test("UserNav logged out", () => {
  const tree = renderer.create(<UserNav cookie={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("UserNav logged in", () => {
  const cookie = {
    username: "samples",
  };
  const tree = renderer.create(<UserNav cookie={cookie} />).toJSON();
  expect(tree).toMatchSnapshot();
});
