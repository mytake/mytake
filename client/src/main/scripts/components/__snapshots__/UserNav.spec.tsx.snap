// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNav logged in 1`] = `
<DropDown
  classModifier="usernav"
  disabled={false}
  dropdownPosition="BL"
  toggleText={
    <React.Fragment>
      <ChevronDown />
      samples
    </React.Fragment>
  }
>
  <ul
    className="usernav__list"
  >
    <li
      className="usernav__list-item"
    >
      <a
        className="usernav__link"
        href="/drafts/new"
      >
        New Draft
      </a>
    </li>
    <li
      className="usernav__list-item"
    >
      <a
        className="usernav__link"
        href="/drafts"
      >
        Drafts
      </a>
    </li>
    <li
      className="usernav__list-item"
    >
      <a
        className="usernav__link"
        href="/samples"
      >
        Published
      </a>
    </li>
    <li
      className="usernav__list-item"
    >
      <a
        className="usernav__link"
        href="/samples?tab=stars"
      >
        Stars
      </a>
    </li>
    <li
      className="usernav__list-item"
    >
      <a
        className="usernav__link"
        href="/samples?tab=edit"
      >
        Profile
      </a>
    </li>
    <li
      className="usernav__list-item"
    >
      <a
        className="usernav__link"
        href="/logout"
      >
        Logout
      </a>
    </li>
  </ul>
</DropDown>
`;

exports[`UserNav logged out 1`] = `
<DropDown
  classModifier="usernav"
  disabled={true}
  dropdownPosition="BL"
  toggleText={
    <a
      href="/login"
    >
      Login
    </a>
  }
>
  <ul
    className="usernav__list"
  />
</DropDown>
`;
